internal/kubernetes/virtualNodeupdater.go

```diff
@@ -55,6 +56,34 @@ func (p *KubernetesProvider) ReconcileNodeFromAdv(event watch.Event) {
 		return
 	}
 
+	if adv.Status.AdvertisementStatus == advertisement_operator.AdvertisementDeleting {
+		for retry := 0; retry < 3; retry++ {
+			klog.Infof("advertisement %v is going to be deleted... set node status not ready", adv.Name)
+			no, err := p.nodeUpdateClient.Client().CoreV1().Nodes().Get(p.nodeName, metav1.GetOptions{})
+			if err != nil {
+				klog.Error(err)
+				continue
+			}
+			for _, condition := range no.Status.Conditions {
+				if condition.Type == v1.NodeReady {
+					no.Status.Conditions[0].Status = v1.ConditionFalse
+					_, err = p.nodeUpdateClient.Client().CoreV1().Nodes().UpdateStatus(no)
```

Comment:
Instead of performing this manual operation, you can update the node status by using the provider's nodeController with [this](https://github.com/LiqoTech/liqo/blob/e0433b9b79574feef7ec9494d65bef7d80127e8a/internal/kubernetes/virtualNodeupdater.go#L145) method. By following this approach you avoid the concurrency problem.

Reply:
thanks a lot!